src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -271,15 +271,6 @@ private static bool RemovalMayIntroduceCastAmbiguity(ParenthesizedExpressionSynt
                 {
                     return true;
                 }
-
-                if (expression.IsKind(SyntaxKind.NumericLiteralExpression))
-                {
-                    var numericLiteral = (LiteralExpressionSyntax)expression;
-                    if (numericLiteral.Token.ValueText.StartsWith("-", StringComparison.Ordinal))
-                    {
-                        return true;
-                    }
-                }
```

Comment:
I removed it because LiteralExpressionSyntax should never start with "-". Expressions like "-5" should always be parsed as a UnaryMinusExpression with an operand that is a LiteralExpressionSyntax "5". But it turns out removing this code fails some tests in the language service. Apparently they manually construct LiteralExpressionSyntax nodes for negative integers and it works somehow. So, for now I reverted this change, and I'm going to file a separate issue for this. Our API should reject attempts to create LiteralExpressionSyntax for negative numbers, and all callers should be updated to invoke it correctly.

/cc @gafter